app/models/link_set.rb

```diff
@@ -1,42 +1,26 @@
 class LinkSet < ActiveRecord::Base
-  include Replaceable
-  include DefaultAttributes
-  include SymbolizeJSON
-
-  validate :links_are_valid
+  has_many :links
 
-  def links=(links_hash)
-    super(links_hash || {})
-  end
+  include Replaceable
 
-private
-  def self.query_keys
-    [:content_id]
-  end
+  def hashed_links
```

Comment:
This method builds a hash of the links and groups them by their link_type.

ie 
```
{
    "organisations" => ["a-uuid-xxx-xxx", "some-other-uuid"]
}
```
In order to do this, the methods you suggested wouldn't suffice but might simplify this method.